backend/compact-connect/tests/unit/test_app.py

```diff
@@ -48,6 +48,29 @@ def _get_resource_properties_by_logical_id(self, logical_id: str, resources: Map
         except KeyError as exc:
             raise RuntimeError(f'{logical_id} not found in resources!') from exc
 
+    def compare_snapshot(self, actual: dict, snapshot_name: str, overwrite_snapshot: bool = False):
```

Comment:
Should we refactor this up to a common root rather than having it defined twice?

Reply:
@jusdino I assumed that was going to be done as part of the test refactor on your branch, so I didn't want to do the same effort twice. But if there is a simple way I can do that without duplicating effort that you want me to do in this PR, let me know.

Reply:
Yeah that works. We'll just keep iterating.